### DISCUSSION TITLE
HotChocolate with Postgres: how to map NpgsqlRange type?

### DISCUSSION THREAD

**Discussion by Arsync** [ChilliCream/hotchocolate#5892] [Q&A] [answered]
I have models with properties of type `NpgsqlRange<>`. When project is started, it throws following errors:

HotChocolate.SchemaException: 'For more details look at the `Errors` property.
1. Unable to infer or resolve a schema type from the type reference `NpgsqlRange<Instant!> (Output)`.
2. Unable to infer or resolve a schema type from the type reference `NpgsqlRange<LocalDate!> (Output)`.

For now I've created graphql model types, where these properties are ignored. But this cannot last forever.

**Comment by PascalSenn** [comment, marked as answer]
@Arsync  the easiest thing is to define a `ObjectType<NpgsqlRange<LocalDate!>>` and `ObjectType<NpgsqlRange<Instant!>>` and register them on the schema.

**Reply by Arsync** [reply]
Thanks! Now I have
```cs
public class InstantRangeType : ObjectType<NpgsqlRange<Instant>>
{
    protected override void Configure(IObjectTypeDescriptor<NpgsqlRange<Instant>> d)
    {
        d.BindFieldsExplicitly();

        d.Field(x => x.LowerBound);
        d.Field(x => x.UpperBound);

        d.Field(x => x.IsEmpty);
    }
}
```
It runs and query looks like
```graphql
query GetPerson($personId: Int!)
{
  person(personId: $personId)
  {
      versionPeriod {
        lowerBound
        upperBound
      }
  }
}
```
But there is exception while executing the query:
> The binary operator NotEqual is not defined for the types 'NpgsqlTypes.NpgsqlRange`1[NodaTime.Instant]' and 'System.Object'.

**Reply by PascalSenn** [reply]
Are you using projection? what version of HC are you on?

**Reply by Arsync** [reply]
Yes, I'm using projection, HC version 13.0.5
When `UseProjectionAttribute` is disabled, query done without any exceptions.
Any chance to use it with projections?
